src/main/java/org/jenkinsci/plugins/github_branch_source/GitHubAppCredentials.java

```diff
@@ -122,11 +126,25 @@ public void setOwner(String owner) {
         this.owner = Util.fixEmpty(owner);
     }
 
+    @SuppressWarnings("deprecation")
+    public AuthorizationProvider getAuthorizationProvider() {
+        // TODO create sub class
+        AuthorizationProvider authorizationProvider;
+        try {
+            authorizationProvider = new JWTTokenProvider(appID, privateKey.getPlainText());
+        } catch (GeneralSecurityException e) {
+            throw new IllegalArgumentException("Couldn't parse private key for GitHub app, make sure it's PKCS#8 format", e);
+        }
+
+        return new OrgAppInstallationAuthorizationProvider(owner, authorizationProvider);
```

Comment:
this auth provider doesn't meet our needs and will need customising but it's enough for initial testing

Reply:
@timja 
Yeah, I figured this would be the case. We'll need an authorization provider that takes the credentials.